--- ../../../../navit/CMakeLists.txt	2014-02-13 21:00:39.461973777 -0800
+++ ../../../../navit/CMakeLists.txt	2014-02-13 16:54:59.660857409 -0800
@@ -120,6 +120,7 @@
 find_package(PythonLibs)
 find_package(OpenSSL)
 find_package(Threads)
+find_package(libspotify REQUIRED)
 libfind_pkg_check_modules(FONTCONFIG fontconfig)
 #Qt detection
 if (NOT DISABLE_QT)
@@ -332,6 +333,10 @@
    set_with_reason(vehicle/gypsy "gypsy lib found" TRUE)
 endif(GYPSY_FOUND)
 
+if (libspotify_FOUND)
+   set_with_reason(plugin/spotify "libspotify found" TRUE)
+endif(LIBSPOTIFY_FOUND)
+
 if (LIBGARMIN_FOUND)
    include_directories(${LIBGARMIN_INCLUDE_DIRS})
    set_with_reason(map/garmin "Garmin library found" TRUE ${LIBGARMIN_LDFLAGS})
@@ -394,6 +399,7 @@
 
 #Modules without test yet
 add_module(plugin/pedestrian "Default" FALSE)
+add_module(plugin/spotify "Default" TRUE)
 add_module(speech/android "Default" FALSE)
 add_module(speech/espeak "Default" FALSE)
 add_module(speech/iphone "Default" FALSE)
