Index: ../../attr_def.h
===================================================================
--- ../../attr_def.h	(revision 5742)
+++ ../../attr_def.h	(working copy)
@@ -376,6 +376,9 @@
 ATTR(last_key)
 ATTR(src_dir)
 ATTR(refresh_cond)
+ATTR(spotify_login)
+ATTR(spotify_password)
+ATTR(spotify_playlist)
 ATTR2(0x0003ffff,type_string_end)
 ATTR2(0x00040000,type_special_begin)
 ATTR(order)
Index: ../../plugin.c
===================================================================
--- ../../plugin.c	(revision 5742)
+++ ../../plugin.c	(working copy)
@@ -134,10 +134,12 @@
 	int lazy;
 	int ondemand;
 	char *name;
+	struct attr *attrs;
 #ifdef USE_PLUGINS
 	GModule *mod;
 #endif
 	void (*init)(void);
+	void (*set_attr)(struct attr *attrs);
 };
 
 struct plugins {
@@ -166,6 +168,7 @@
 {
 #ifdef USE_PLUGINS
 	gpointer init;
+	gpointer set_attr;
 
 	GModule *mod;
 
@@ -187,6 +190,13 @@
 		pl->mod=mod;
 		pl->init=init;
 	}
+	if (!g_module_symbol(mod, "plugin_set_attr", &set_attr)) {
+		dbg(0,"plugin %s does not provide plugin_set_attr\n", pl->name);
+		pl->set_attr=NULL;
+	} else {
+		dbg(0, "plugin %s does provide plugin_set_attr\n", pl->name);
+		pl->set_attr=set_attr;
+	}
 	return 1;
 #else
 	return 0;
@@ -231,10 +241,23 @@
 	pl->ondemand=ondemand;
 }
 
+static void
+plugin_set_attrs(struct plugin *pl, struct attr *attrs)
+{
+	pl->attrs=attrs;
+	dbg(0,"attrs set for plugin %s with size %lu\n", pl->name, sizeof(attrs));
+}
+
 void
 plugin_call_init(struct plugin *pl)
 {
 	pl->init();
+	if(pl->set_attr){
+		dbg(0,"*** calling set attrs for %s with sizeof %lu\n", pl->name, pl->attrs);
+		pl->set_attr(pl->attrs);
+	} else {
+		dbg(0,"*** pl->attrs is NULL for %s\n", pl->name);
+	}
 }
 
 void
@@ -290,7 +313,7 @@
 	if ( (attr=attr_search(attrs, NULL, attr_ondemand))) {
 		ondemand=attr->u.num;
 	}
-	dbg(1, "path=\"%s\", active=%d, lazy=%d, ondemand=%d\n",path_attr->u.str, active, lazy, ondemand);
+	dbg(0, "path=\"%s\", active=%d, lazy=%d, ondemand=%d\n",path_attr->u.str, active, lazy, ondemand);
 
 	we=file_wordexp_new(path_attr->u.str);
 	count=file_wordexp_get_count(we);
@@ -319,6 +342,7 @@
 			plugin_set_active(pl, active);
 			plugin_set_lazy(pl, lazy);
 			plugin_set_ondemand(pl, ondemand);
+			plugin_set_attrs(pl, attrs);
 			if (!pls && active) {
 				if (!plugin_load(pl)) 
 					plugin_set_active(pl, 0);
